linalg/invariants.hpp

```diff
@@ -97,6 +97,9 @@ class InvariantsEvaluator2D
       eval_state |= HAVE_I2b;
       const scalar_t det = J[0]*J[3] - J[1]*J[2];
       sign_detJ = scalar_ops::sign(det);
+
+      sign_detJ = 1.0;
+
       I2b = sign_detJ*det;
```

Comment:
Just set `I2b = det;` here and save a couple flops? We could leave a comment for future us to indicate that if we want `I2b` to be positive, we need to multiply with the `sign(I2b)`.